### PR TITLE
many: implement snap alias <snap.app> <alias> (aliases v2)

### PR DESCRIPTION
**Pull request by pedronis**
Q:  snap alias SNAP-NAME ALIAS   (to alias the main app of a snap is ambiguous with the old command working for enabling one predefined alias)

wondering if  snap alias ALIAS SNAP.APP  would be a better ordering of arguments because of that

### CODE REVIEW

**Comment by pedronis** [comment]
based on #3137

**Comment by niemeyer** [comment]
> Q: snap alias SNAP-NAME ALIAS (to alias the main app of a snap is ambiguous with the old command working for enabling one predefined alias)

That sounds okay. Let's just make sure the command output reflects well what happened. We've already decided to break CLI-level compatibility for this feature in the name of improving the overall semantics and behavior, so this is just one small inconvenience of that.

File: client/aliases.go [hunk 1]
```diff
@@ -26,9 +26,12 @@ import (
 
 // aliasAction represents an action performed on aliases.
 type aliasAction struct {
-	Action  string   `json:"action"`
-	Snap    string   `json:"snap"`
-	Aliases []string `json:"aliases"`
+	Action string `json:"action"`
+	Snap   string `json:"snap"`
+	Target string `json:"target,omitempty"`
```
Comment (niemeyer) [inline client/aliases.go]: Wouldn't that be more clear as `"App"/"app"`? Looking at the rest of the PR, it feels like that's what we talk about everywhere else.

Comment (pedronis) [inline client/aliases.go]: the GET /aliases previous API was using "app"  for "snap.app"  , wondering if that would make this confusing? (this is really just the app without the snap part), I think the old API was probably wrong doing that

Comment (niemeyer) [inline client/aliases.go]: Yeah, sounds like it was bogus before.

**Pull request merged**